src/actions/write/saveObservations.ts

```diff
@@ -35,9 +38,14 @@ export class SaveObservations {
         ),
       );
     }
-    const { observerReportTxId, failedGateways } = input;
+    const {
+      observerReportTxId,
+      failedGateways,
+      gatewayAddress = caller, // default the gateway address to the caller
+    } = input;
     this.observerReportTxId = observerReportTxId;
     this.failedGateways = failedGateways;
+    this.gatewayAddress = gatewayAddress;
```

Comment:
note: the ar-io-observer will need to account for submitting observations against gateway addresses that do not match the observer wallet address cc @djwhitt

Reply:
nit: this could be an array if we want to handle that.

Reply:
another thought - enforce uniqueness on observerwallets

Reply:
TLDR - we're going with enforcing unqiueness